CONTRIBUTING.md

```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
+
+```
+cd /[path to my code folder]/astro
 ```
 
+Next, you’ll need to install dependencies and set up the project. You can do that by running the following commands from within the `astro/` folder:
+
+```
+yarn
+yarn build
+```
+
+This will download everything you need and build Astro locally for use. You’re ready to go!
+
 ## Development
 
-```shell
-# starts a file-watching, live-reloading dev script for active development
+While working within packages (such as `packages/astro/*`), you may notice you have to run `yarn build` to see your changes. This process can really slow you down if you are trying many changes at once. Try running the following instead:
+
+```
 yarn dev
-# build the entire project, one time.
+```
+
+This will start a build watcher, which will rebuild files as you work. Use this to speed up your workflow, and see your changes more rapidly.
+
+Of course, you can still build all files within a package, as before. You may need to do so from time to time:
+
+```
 yarn build
 ```
 
 ## Running tests
 
-```shell
-# run this in the top-level project root to run all tests
+You’ll need to run existing tests to make sure your work didn’t cause any regressions. To create a fresh build and run all tests, run the following from the project root:
+
+```
+yarn
+yarn build
 yarn test
-# run only a few tests, great for working on a single feature
-# (example - `yarn test rss` runs `astro-rss.test.js` tests)
-yarn test $STRING_MATCH
 ```
 
+If you find yourself wanting to not run the whole test suite, you can run the following instead:
+
+```
+cd packages/astro
+yarn test astro-markdown
+```
+
+Replace “`astro-markdown`“ with any part of a test filename to only run those tests.
+
+## Making a Pull Request
+
+When you’re ready to commit your work and submit it for review, you’ll need to do the following:
+
+1. Create a new **branch**:
+   ```
+   git checkout -b fix-homepage-bug
+   ```
+1. **Commit** your changes:
+   ```
+   git add *
+   git commit -m "Fix layout bug on homepage"
+   ```
+   _Note: be short and specific in your message! This will live forever!_
+1. **Push** your changes:
+   ```
+   git push --set-upstream origin fix-homepage-bug
+   ```
+1. Add a **changeset** message:
+
+   ```
+   yarn changeset
+   ```
+
+   Your changeset message will be what appears publicly in release notes. It’s all right if it’s more verbose than your commit message.
+
+   Also, while Astro is pre-1.0, you probably want to only make **patch** changes, unless you’re releasing a huge, sweeping, breaking change as a **minor** release ([what is semver?][semver]).
+
+1. [Open a new Pull Request][astro-pr] on Astro with your changes.
+
+Be sure to **fill in the template** completely! It helps other people review your work.
+
+If you need to create a pull request to get early feedback, that’s OK, too.
+
+Creating your first Pull Request can seem like a lot of steps at first, but with time and practice it’ll become comfortable.
```

Comment:
fyi this is what made me think of the "my 2 cents" comment above. If we reduced this section to just: "Here's a great GitHub-written and reviewed guide on how to submit your first Pull Request" I think we would get more contributions and a lot fewer "hey, I did X from the docs and it didn't work" support requests.